src/main/java/it/gov/pagopa/pu/debtpositions/service/create/debtposition/DebtPositionProcessorServiceImpl.java

```diff
@@ -18,8 +20,25 @@ public void updateAmounts(DebtPositionDTO debtPositionDTO) {
     });
   }
 
+  @Override
+  public void updateAmounts(DebtPosition debtPosition) {
+    debtPosition.getPaymentOptions().forEach(paymentOption -> {
+      long totalPaymentOptionAmount = paymentOption.getInstallments().stream()
+        .filter(this::isInstallmentCancelled)
+        .mapToLong(InstallmentNoPII::getAmountCents)
+        .sum();
+
+      paymentOption.setTotalAmountCents(totalPaymentOptionAmount);
+    });
+  }
+
   private boolean isInstallmentCancelled(InstallmentDTO installment){
     return !(installment.getStatus() == InstallmentStatus.CANCELLED ||
       (installment.getSyncStatus() != null && InstallmentStatus.CANCELLED.equals(installment.getSyncStatus().getSyncStatusTo())));
   }
+
+  private boolean isInstallmentCancelled(InstallmentNoPII installment){
+    return !(installment.getStatus() == InstallmentStatus.CANCELLED ||
+      (installment.getSyncStatus() != null && InstallmentStatus.CANCELLED.equals(installment.getSyncStatus().getSyncStatusTo())));
+  }
```

Comment:
let's not copy the method, but let's just change the required parameter asking for a `BaseInstallment` instead of InstallmentDTO